### ISSUE TITLE
Network Profiles mangement (Bugzilla Bug 14)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#14]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 14
Date: 2013-06-23 13:32:19 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;
To:   OpenMandriva QA &lt;<bugs@openmandriva.org>&gt;
CC:   agoiza@gmail.com, arisel@arisel.de, @benbullard79, @cris-b, nix.or.die@gmail.com

Depends on:   #2093
Last updated: 2018-06-21 23:55:49 +0000

## Comment 39
Date: 2013-06-23 13:32:19 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Network profiles are very useful in all portable devices that must be connected to several different networks. One of the jewel in the crown of Mandriva.

It would be useful an icon or any shortcut to have a simple way to know which profile is active and to change it if necessary. The best way could be show active profile and provide a way to change it in Network manager window. 

In previous versions of Mandriva (till 2010.2 ?) we had netapplet in systray. Restoring it could be a faster alternative.

## Comment 112
Date: 2013-06-25 22:27:08 +0000
From: @robxu9

[this is a platform change to ensure correctness. if the platform is not correct after this change, please fix it. possible valid values are armv7hl_tgr2, armv7nhl_omp3, i586, x86_64]

## Comment 121
Date: 2013-06-26 04:49:55 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

I asked it for x86_64 because I use this platform but this feature is useful in both platform x86_64 and i586

## Comment 122
Date: 2013-06-26 06:40:15 +0000
From: arisel &lt;<arisel@arisel.de>&gt;

ok, changed it to ALL/enhancement

## Comment 4941
Date: 2014-04-06 06:44:21 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Now netprofile is part of the distribution and works well.

1) Is it possible to show the active profile in network manager ?

2) Allow an easy way to change profile. Is it possible to add a "kcm" applet to open netprofile window? 
A good example is in MIB repos http://mib.pianetalinux.org/MIB/rosa2012.1/64/basic/kcm-draknetprofile-1.0-69.2-mib2012.1.noarch.rpm

3) As soon as the first netprofile is created add an item in menu to open netprofile window

## Comment 16172
Date: 2017-03-20 10:18:25 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

The request is still valid for OpenMandriva Lx 3.

## Comment 16174
Date: 2017-03-20 12:40:02 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

Hope is the last to die ;)

## Comment 16206
Date: 2017-03-26 18:00:25 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

take a look at netctl and netctl-gui.

if ok you could request for rpm.

## Comment 16275
Date: 2017-04-09 10:07:26 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Thanks for suggestion. It seems promising but it I don't find any image of GUI just to understand which features it can affect. Anyway could be a solution.

Asked for packaging in bug #2093.

## Comment 17938
Date: 2018-03-05 23:38:24 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

After closing bug #2093 a simple icon could be a temporary solution.

## Comment 17944
Date: 2018-03-06 02:56:07 +0000
From: Gabriel Craciunescu &lt;<nix.or.die@gmail.com>&gt;

(In reply to Giorgio from comment #9)
> After closing bug #2093 a simple icon could be a temporary solution.

What do you mean by an 'icon' ?

## Comment 17954
Date: 2018-03-06 17:05:46 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

To change netprofile you have to open MCC then choose network and finally Netprofile. It would be useful to have an icon in menu that open Netprofile window directly. We had it several years ago. 
What I wrote in Description.

## Comment 18198
Date: 2018-06-03 16:06:48 +0000
From: @benbullard79

This utility has been superseded by NetworkManager. You can easily create network profiles in it. So the icon you seek exists it is called NetworkManager applet.

This is actually easy. Open the NetworkManager applet and click on the plus sign and create what ever you want. You can even assign certain users to a given connection. If users will spend maybe 5 or 10 minutes reading the NetworkManager documentation there is much, more you can do.

https://wiki.gnome.org/Projects/NetworkManager

FWIW: The person who wrote the specific application you are talking about is gone, not with OpenMandriva. As with the vast majority of Linux distros in 2018 network functions at the user and administrator level are handled by NetworkManager. OpenMandriva developers tell me this is not going to change anytime soon.

## Comment 18199
Date: 2018-06-03 16:09:50 +0000
From: @benbullard79

The NetworkManager applet is also called the NM-connections editor. From:

https://help.ubuntu.com/community/NetworkManager

NM-connections editor

The NM-connections editor is where the user can edit and create the various connection types. To invoke the nm-connections editor you need to right-click on the nm-applet and then click on Edit Connections. Clicking on that option would result in a tabbed window called network connections having different ways to connect to the network/Internet.

## Comment 18313
Date: 2018-06-21 22:26:51 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to Ben Bullard from comment #12)
> This utility has been superseded by NetworkManager. You can easily create
> network profiles in it. So the icon you seek exists it is called
> NetworkManager applet.
Netprofile can change in one shot proxy settings, urpmi settings, firewall, services and network connections? Especially useful if you use (like me) a notebook in different networks.
Netprofile affect all these settings? No! 

> This is actually easy. Open the NetworkManager applet and click on the plus
> sign and create what ever you want. You can even assign certain users to a
> given connection. If users will spend maybe 5 or 10 minutes reading the
> NetworkManager documentation there is much, more you can do.
> 
> https://wiki.gnome.org/Projects/NetworkManager
Thanks. I use normally networkmanager to manage networks connections (not proxy nor urpmi settings).

> FWIW: The person who wrote the specific application you are talking about is
> gone, not with OpenMandriva. As with the vast majority of Linux distros in
> 2018 network functions at the user and administrator level are handled by
> NetworkManager. OpenMandriva developers tell me this is not going to change
> anytime soon.
Then developers too don't know netprofile features? 

If you want you can close this bug as "cantfix" but this isn't invalid.

## Comment 18315
Date: 2018-06-21 23:55:49 +0000
From: @benbullard79

Well OK then. 

I can see the usefulness of this software but only for a small handful of users at this time. You aren't even close to the first person in computer science that has seem some useful, even very useful, software go away.

OpenMandriva developers are not even close to the best group to keep asking to resolve this. There are not enough of them. At this time there is no interest in helping system administrators do things with a GUI that can already be done from command line.

You could file an upstream bug report as a feature request with NetworkManager project. 

Also you could do some research and see if you can find software that does what you want in Linux/Unix. Then if such software exists the possibility of importing it can be explored. Perhaps look first at Mageia, specifically Mageia's Manatools. There is discussion that Manatools be imported to OpenMandriva for Lx 4 release as a replacement for drakxtools. (That's discussion at this point, not definite.) But do look else where in Linux/Unix World as well.

Our developers keep insisting to me that you absolutely can do what you want in NetworkManager. No one says you can with one click. But it is doable and they say lots of folks do it. Further they tell me anyone who administers multiple computers that other people use should be able to do this with a little learning.
